NewHorizons/Builder/Props/GeneralPropBuilder.cs

```diff
@@ -12,65 +12,85 @@ namespace NewHorizons.Builder.Props
 {
     public static class GeneralPropBuilder
     {
-        public static GameObject MakeFromExisting(GameObject go, GameObject planetGO, Sector sector, PropModule.PositionedPropInfo info, bool alignToBody = false, MVector3 normal = null)
+        public static GameObject MakeFromExisting(GameObject go, Transform parent, PropModule.GeneralPointPropInfo info, bool alignToBody = false, MVector3 normal = null, MVector3 defaultPosition = null, string defaultParentPath = null)
         {
+            if (info == null) return go;
+
+            if (info is PropModule.GeneralSolarSystemPropInfo solarSystemInfo && !string.IsNullOrEmpty(solarSystemInfo.parentBody))
+            {
+                var targetPlanet = AstroObjectLocator.GetAstroObject(solarSystemInfo.parentBody);
+                if (targetPlanet != null)
+                {
+                    parent = targetPlanet.transform;
+                } else
+                {
+                    Logger.LogError($"Cannot find parent body named {solarSystemInfo.parentBody}");
+                }
+            }
+
             if (!string.IsNullOrEmpty(info.rename))
             {
                 go.name = info.rename;
             }
 
-            go.transform.parent = sector?.transform ?? planetGO.transform;
+            go.transform.parent = parent;
 
-            if (!string.IsNullOrEmpty(info.parentPath))
+            var parentPath = info.parentPath ?? defaultParentPath;
+
+            if (!string.IsNullOrEmpty(parentPath))
             {
-                var newParent = go.transform.Find(info.parentPath);
+                var newParent = go.transform.Find(parentPath);
                 if (newParent != null)
                 {
                     go.transform.parent = newParent.transform;
                 }
                 else
                 {
-                    Logger.LogError($"Cannot find parent object at path: {go.name}/{info.parentPath}");
+                    Logger.LogError($"Cannot find parent object at path: {go.name}/{parentPath}");
                 }
             }
 
-            Vector3 pos = (Vector3)(info.position ?? Vector3.zero);
-            Quaternion rot = Quaternion.identity;
-            if (info is PropModule.PositionedAndRotatedPropInfo rotInfo)
+            var pos = (Vector3)(info.position ?? defaultPosition ?? Vector3.zero);
+            var rot = Quaternion.identity;
+            if (info is PropModule.GeneralPropInfo rotInfo)
             {
                 rot = rotInfo.rotation != null ? Quaternion.Euler(rotInfo.rotation) : Quaternion.identity;
             }
             if (info.isRelativeToParent)
             {
                 go.transform.localPosition = pos;
                 go.transform.localRotation = rot;
+            } else if (parent)
+            {
+                go.transform.position = parent.root.TransformPoint(pos);
```

Comment:
does doin this relative to the sector vs planet instead of always planet break backwards compat?

Reply:
Good catch, I'll test that. For custom planets the sector's always at (0,0,0) so it should make a difference but I guess it might for vanilla planets.

Reply:
On a second look, nope, this should always be planet (`parent.root` should grab the root body, not the sector).